ccm_web/client/src/api.ts

```diff
@@ -1,36 +1,63 @@
+import { CanvasCourseBase } from './models/canvas'
 import { Globals, HelloMessageData } from './models/models'
 import handleErrors from './utils/handleErrors'
 
 export interface LtiProps {
   ltiKey: string | undefined
 }
 
-const createAuthHeaders = (key: string | undefined): RequestInit => {
+const initRequest = (key: string | undefined, headers: string[][] = []): RequestInit => {
```

Comment:
In Python, using a mutable default parameter is an antipattern (I've been corrected on this before), and the recommended remedy is using a sentinel value. Not sure if it's an issue in JavaScript.

https://docs.quantifiedcode.com/python-anti-patterns/correctness/mutable_default_value_as_argument.html

Reply:
Hmm, I don't think that is how javascript works.. in python that creates some sort of  global variable?

Reply:
Yeah, I didn't see anything about this being a problem in JS, but I thought I'd flag it in case you knew something about it. I don't think it would be wrong to use `null` here as the default, but it may not be necessary.